test/LiveDeploy.t.sol

```diff
@@ -0,0 +1,359 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity 0.8.21;
+
+import { Test, stdStorage, StdStorage, stdError, console } from "@forge-std/Test.sol";
+import { DeployAll } from "script/deploy/deployAll.s.sol";
+import { ConfigReader } from "script/ConfigReader.s.sol";
+import { ERC20 } from "@solmate/tokens/ERC20.sol";
+import { BoringVault } from "src/base/BoringVault.sol";
+import { FixedPointMathLib } from "@solmate/utils/FixedPointMathLib.sol";
+
+import { TellerWithMultiAssetSupport } from "src/base/Roles/TellerWithMultiAssetSupport.sol";
+import { AccountantWithRateProviders } from "src/base/Roles/AccountantWithRateProviders.sol";
+import { RolesAuthority } from "@solmate/auth/authorities/RolesAuthority.sol";
+import { DeployRateProviders } from "script/deploy/01_DeployRateProviders.s.sol";
+
+import { CrossChainOPTellerWithMultiAssetSupportTest } from
+    "test/CrossChain/CrossChainOPTellerWithMultiAssetSupport.t.sol";
+import { CrossChainTellerBase, BridgeData, ERC20 } from "src/base/Roles/CrossChain/CrossChainTellerBase.sol";
+import { CrossChainOPTellerWithMultiAssetSupport } from
+    "src/base/Roles/CrossChain/CrossChainOPTellerWithMultiAssetSupport.sol";
+import { MultiChainLayerZeroTellerWithMultiAssetSupport } from
+    "src/base/Roles/CrossChain/MultiChainLayerZeroTellerWithMultiAssetSupport.sol";
+
+string constant DEFAULT_RPC_URL = "L1_RPC_URL";
+
+// We use this so that we can use the inheritance linearization to start the fork before other constructors
+abstract contract ForkTest is Test {
+    constructor() {
+        _startFork(DEFAULT_RPC_URL);
+    }
+
+    function _startFork(string memory rpcKey) internal virtual returns (uint256 forkId) {
+        if (block.chainid == 31_337) {
+            forkId = vm.createFork(vm.envString(rpcKey));
+            vm.selectFork(forkId);
+        }
+    }
+}
+
+contract LiveDeploy is ForkTest, DeployAll {
+    using FixedPointMathLib for uint256;
+
+    ERC20 constant NATIVE_ERC20 = ERC20(0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE);
+    uint256 ONE_SHARE;
+    uint8 constant SOLVER_ROLE = 42;
+
+    function setUp() public virtual {
+        string memory FILE_NAME;
+        if (block.chainid == 31_337) {
+            FILE_NAME = "exampleL1.json";
+        } else {
+            FILE_NAME = vm.envString("LIVE_DEPLOY_READ_FILE_NAME");
+        }
+
+        // we have to start the fork again... I don't exactly know why. But it's a known issue with foundry re:
+        // https://github.com/foundry-rs/foundry/issues/5471
+        _startFork(DEFAULT_RPC_URL);
```

Comment:
Because this test inherits the DeployAll contract which inherits Base, and base script has a constructor that attempts to access CREATEX and will fail if not on a fork